spec/models/sitemap_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'spec_helper'
+
+describe Sitemap do
+  let!(:url) { 'http://www.agency.gov/boring' }
+  let!(:html) { read_fixture_file("/html/page_with_og_metadata.html") }
+  let!(:valid_attributes) { { url: url } }
+  let!(:sitemap) { Sitemap.new(valid_attributes) }
+  let!(:existing_domain) { SearchgovDomain.create!(domain: 'existing.gov') }
+
+  it { is_expected.to have_readonly_attribute(:url) }
+
+  describe 'schema' do
+    it { is_expected.to have_db_column(:url).of_type(:string).with_options(null: false, limit: 2000) } # causing problems for I'm not sure why
+    it { is_expected.to have_db_column(:last_crawl_status).of_type(:string) }
+    it { is_expected.to have_db_column(:last_crawled_at).of_type(:datetime) }
+  end
+
+  describe 'associations' do
+    it { is_expected.to belong_to(:searchgov_domain) }
+
+    context 'on creation' do
+      context 'when the domain already exists' do
+        it 'sets the searchgov domain' do
+          sitemap = Sitemap.create!(url: 'https://existing.gov/foo.xml')
+          expect(sitemap.searchgov_domain).to eq(existing_domain)
+        end
+      end
+
+      context 'when the domain has not been created yet' do
+        it 'creates the domain' do
+          expect{ Sitemap.create!(url: 'https://brandnewdomain.gov/loop.txt') }.
+            to change{ SearchgovDomain.count }.by(1)
+        end
+      end
+    end
+  end
+
+  describe 'validations' do
+    it 'requires a valid domain' do
+      sitemap = Sitemap.new(url: 'https://foo/bar')
+      expect(sitemap).not_to be_valid
+    end
+
+    context 'when validating url uniqueness' do
+      let!(:existing) { Sitemap.create!(valid_attributes) }
+
+      it { is_expected.to validate_uniqueness_of(:url).on(:create) }
+
+      it 'is case-sensitive' do
+        expect(Sitemap.new(url: 'https://www.agency.gov/BORING.xml')).to be_valid
+      end
+    end
+
+    context 'when validating url presence (not being nil)' do
+      let!(:attempt) { Sitemap.new }
+
+      it { is_expected.not_to be_valid }
+    end
+
+    context 'when validating last_crawl_status when it is > 255 characters' do
```

Comment:
As this `before_validation` check is in `Fetchable`, this spec should be in the shared examples for `'a record with a fetchable url'`.

It's best to use a single condition for a context; you can always nest multiple contexts. That keeps your specs flexible, as you may want to add additional nested contexts within the same parent context. And you should make sure that the conditions for your context are set in a `let` or `before` block, rather than in the spec itself. In this case, I would do something like:
```ruby
context 'when the last_crawl_status is > 255 characters' do
  let(:sitemap) { Sitemap.new(valid_attributes.merge(last_crawl_status: 'x' * 300) }

  it 'truncates the last_crawl_status' do
    expect(sitemap.valid?).to eq true
    expect(sitemap.last_crawl_status).to eq 'x' * 255
  end
end
```
(Note: normally I wouldn't use two `expect` lines in a single spec, but in this case I would need to call `sitemap.valid?` anyway before checking the `last_crawl_status` value, so making separate specs would be somewhat redundant.)